game.js

```diff
@@ -39,50 +39,31 @@ class Game {
         this.socket.on(`game/${game_id}/gamedata`, (gamedata) => {
             if (!this.connected) return;
 
-            //this.log("Gamedata:", JSON.stringify(gamedata, null, 4));
-
-            const prev_phase = (this.state ? this.state.phase : null);
-            this.state = gamedata;
-            this.my_color = this.conn.bot_id === this.state.players.black.id ? "black" : "white";
-            this.log(`gamedata     ${this.header()}`);
-
-            this.conn.addGameForPlayer(gamedata.game_id, this.getOpponent().id);
-
             // Only call game over handler if game really just finished.
             // For some reason we get connected to already finished games once in a while ...
-            if (gamedata.phase === 'finished' && prev_phase && gamedata.phase !== prev_phase)
-                this.gameOver();
+            if (gamedata.phase === 'finished') {
+                const prev_phase = (this.state ? this.state.phase : null);                
+                this.state = gamedata;                
+                if (prev_phase && gamedata.phase !== prev_phase) this.gameOver();
+                return; // ignore if we already handled it
```

Comment:
The comment here doesn't seem to fit. Did you maybe mean to put this before the prior if() line?

I'm not sure which is more clean. How you do this now, or more like

if (gamedata.phase === 'finished') {
    if (this.state && gamedata.phase !== this.state.phase) {
        this.state = gamedata;
        this.gameOver();
    }
    return;
}

Just to shorten it, remove temp variable, and not assign to this.state if not needed. Not a bit deal.

Reply:
Maybe it's better to assign every time though so this.state is always up to date. Even if we think we'll never need it today.

Reply:
done